vmhost/vmhooks/manBufOps.go

```diff
@@ -542,11 +542,15 @@ func (context *VMHooksImpl) MBufferStorageLoad(keyHandle int32, destinationHandl
 		return 1
 	}
 
-	storageBytes, usedCache, err := storage.GetStorage(key)
+	storageBytes, trieDepth, usedCache, err := storage.GetStorage(key)
 	if context.WithFault(err, runtime.ManagedBufferAPIErrorShouldFailExecution()) {
 		return 0
 	}
-	storage.UseGasForStorageLoad(mBufferStorageLoadName, metering.GasSchedule().ManagedBufferAPICost.MBufferStorageLoad, usedCache)
+
+	err = storage.UseGasForStorageLoad(mBufferStorageLoadName, trieDepth, metering.GasSchedule().ManagedBufferAPICost.MBufferStorageLoad, usedCache)
+	if context.WithFault(err, runtime.BaseOpsErrorShouldFailExecution()) {
+		return -1
```

Comment:
In this case, it's an error value, but it will never be received by the WASM code because `context.WithFault()` sets a hard breakpoint if `err` is not nil. The actual return value is irrelevant, but a return is needed to trigger the breakpoint right then.

However, we still need to simplify and clean this up in **_all the VM hooks_** (planned).

Reply:
There are various values used because some VM hooks had to return something that made sense, like a 0 length of something (even if the contract was stopped before seeing the 0 value). -1 was used as an error value if 1 could be a productive value, but in some cases the hook had to return unsigned values.

In other cases, 0 means OK and 1 means error. It's a mix of return values, unfortunately, but it will be cleaned up.

Reply:
Ok, if is planned to be fixes, it's ok. We can have defined types over a byte or int to define some error types

Reply:
Yes, the core problem is the lack of named constant definitions. Otherwise, returning 0, -1 or anything else is fine; the meaning will depend on the function itself.